### PR TITLE
Add CHANGELOG.md

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

This PR proposes adding a changelog to track notable changes in releases.

Changes were tested by locally tagging a commit with `r3.9.0` and temporarily tweaking the script to produce a draft. Here is a sample draft of the produced release notes:
https://github.com/mongodb/mongo-cxx-driver/releases/tag/untagged-fd8ab6ba4086415ac8fa

# Background & Motivation
Having a CHANGELOG may reduce work during releases:
- The [release notes for 3.8.0](https://github.com/mongodb/mongo-cxx-driver/releases/tag/r3.8.0) included written notes not generated by `make_release.py`. Notes can be written incrementally with the changes.
- Tickets requiring release notes can be resolved before a release. Example: CXX-2662.

Listing individual Jira tickets is intentionally removed. IMO a copied list of tickets does not provide much value over linking to Jira. Providing a high level overview of changes may be more helpful. Readers can follow the Jira link or see commits for details.

### CODE REVIEW

**Review by rcsanchez97 — COMMENTED** [review]
This reminds of a nicer version of csv2cl and svn2cl. I like the utility of this for users and I also like that it doesn't just reproduce the git commit history (since lots of changes aren't interesting to users).

Do you need an approval so that you can merge this, or are you still thinking about other related changes in this PR or something like that?

**Comment by kevinAlbs** [comment]
> Do you need an approval so that you can merge this, or are you still thinking about other related changes in this PR or something like that?

I am not planning on making more changes. At least one approval is needed to merge. I plan to ask for opinions in the team channel again before merging.

**Comment by rcsanchez97** [comment]
> > Do you need an approval so that you can merge this, or are you still thinking about other related changes in this PR or something like that?
> 
> I am not planning on making more changes. At least one approval is needed to merge. I plan to ask for opinions in the team channel again before merging.

OK. I'll add a LGTM so that you can merge when you're ready.

**Review by rcsanchez97 — APPROVED** [review]

**Review by eramongodb — APPROVED** [review]

**Pull request merged**
